### PR TITLE
enhance: allow vortex reader use the logical size to split chunks

### PR DESCRIPTION
**Pull request by jiaqizho**
This commit adds support for the vortex chunk reader to split chunks 
using the logical chunk size (PROPERTY_READER_VORTEX_CHUNK_ROWS). 
Note that the final chunk size may not be equal to the logical size. 
because that vortex splits chunks according to the layout size by 
default, so when the logical chunk size exceeds the layout size,
the final chunk size becomes the layout size.

Additionally, this PR fixes an issue where `arrow::utf8` is read 
as `arrow::utf8view`. The vortex performs canonical encoding 
operations and converts the arrow schema according to the rules
`in vortex-dtype/src/arrow.rs` by default. Therefore, in the 
format reader, it is necessary to pass an `arrow::schema` to
ensure read type are consistent with write type.

### CODE REVIEW

**Comment by shaoting-huang** [comment]
Can we cover some edge case tests? Like:
* row_start == row_end
* row_end > files total rows
* handling empty files

File: cpp/include/milvus-storage/properties.h [hunk 1]
```diff
@@ -107,6 +107,7 @@ struct PropertyInfo {
 // --- Define Reader property keys ---
 #define PROPERTY_READER_RECORD_BATCH_MAX_ROWS "reader.record_batch_max_rows"
 #define PROPERTY_READER_RECORD_BATCH_MAX_SIZE "reader.record_batch_max_size"
+#define PROPERTY_READER_VORTEX_CHUNK_ROWS "reader.vortex.chunk_rows"
```
[the remark this answers is not in the document]
Comment (jiaqizho) [reply cpp/include/milvus-storage/properties.h]: added in `REGISTER_PROPERTY(PROPERTY_READER_VORTEX_CHUNK_ROWS`
